python/cudf/cudf/core/dataframe.py

```diff
@@ -2638,31 +2695,149 @@ def set_index(self, index, drop=True):
             str : name of column to be used as series
             list of str : name of columns to be converted to a MultiIndex
         drop : boolean
-            whether to drop corresponding column for str index argument
+            Whether to drop corresponding column for str index argument
+        append : boolean
+            Append current to new index to form a multiindex
+        inplace : boolean
+            Modify the DataFrame in place (do not create a new object)
+        verify_integrity : boolean
+            Check for duplicates in the new index
+
+        Examples
+        --------
+        Set existing column as index
+        >>> df = cudf.DataFrame({"a": [1,2,3,4,5],
+        ... "b":["a", "b", "c", "d","e"],
+        ... "c":[1.0, 2.0, 3.0, 4.0, 5.0]})
+
```

Comment:
```suggestion
```

Sphinx will create a new code block because of this empty line, we don't want that in this case.